Python: make tool call view optional in DevUI + other link fixes

victordibia opened the pull request:
### Motivation and Context

Currently tool/function call requests were prominently shown in devui agent chat view even though the same information was already present in the debug panel. this pr creates a smaller view for tool calls (also show detail on demand) to reduce clutter and a UI settings toggle to not show it as at al. 

PR extended to address
-  fix #2351 , ensure concurrent agent messages are shown correctly 
- fix #2334 - ensure multiple HIL messages are shown in workflow not just the first 
- fix  #2310 - ensure the right backend url is shown

<img width="1728" height="1095" alt="image" src="https://github.com/user-attachments/assets/09cafbf6-069f-4923-8b1f-efb6a87a7457" />
<img width="1724" height="1098" alt="image" src="https://github.com/user-attachments/assets/8886ed4d-2496-47f0-b0bf-30ec2fb136d8" />
<img width="1728" height="1095" alt="image" src="https://github.com/user-attachments/assets/dde80fec-9ad8-41ce-8a4e-625fa1519a4d" />

This PR also fixes an error where `responses.completed` was emitted twice during a run. 

<!-- Thank you for your contribution to the Agent Framework repo!
Please help reviewers and future users, providing the following information:
  1. Why is this change required?
  2. What problem does it solve?
  3. What scenario does it contribute to?
  4. If it fixes an open issue, please link to the issue here.
-->

### Description

<!-- Describe your changes, the overall approach, the underlying design.
     These notes will help understanding how your code works. Thanks! -->

### Contribution Checklist

<!-- Before submitting this PR, please make sure: -->

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/agent-framework/blob/main/CONTRIBUTING.md)
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] **Is this a breaking change?** If yes, add "[BREAKING]" prefix to the title of the PR.

victordibia committed:
make tool call view optional in devui + other link fixes

markwallace-microsoft commented:
<!-- Pytest Coverage Comment: python-test-coverage-report -->
<a href="https://github.com/microsoft/agent-framework/blob/af5dfa22470e627f8a41e83bc2aa596de52dfe10/README.md"><img alt="Python Test Coverage" src="https://img.shields.io/badge/Python Test Coverage-85%25-green.svg" /></a><details><summary>Python Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>14965</b></td><td><b>2224</b></td><td><b>85%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 2039 | 129 :zzz: | 0 :x: | 0 :fire: | 40.856s :stopwatch: |

victordibia committed:
Merge branch 'main' into fix_tools_view_devui

victordibia committed:
Merge remote-tracking branch 'origin/main' into fix_tools_view_devui

victordibia committed:
fix #2310, ensure correct port is shown in command

victordibia committed:
fix dialog bug

victordibia committed:
ensure executor ids are tracked per items, fix bug where data from concurrent executors where not seperated properly fix #2351

victordibia committed:
fix: Enable multi-round human-in-the-loop (HIL) in DevUI workflows

- Backend: Enrich RequestInfoEvents with response schemas in send_responses_streaming path
- Frontend: Replace old HIL requests with new ones instead of accumulating them
- Frontend: Fix HIL response state management to prevent sending stale request responses

This allows workflows to properly handle sequential HIL requests, showing only the
current request to users and progressing through multiple input rounds correctly.

fixes #2334

victordibia committed:
fix bug to ensure in memory entities cannot be reloaded in ui
